lib/play_game.rb

```diff
@@ -5,60 +5,73 @@ class PlayGame
 
   attr_reader :player_board,
               :computer_board,
-              :player_cruiser,
-              :player_submarine,
+              # :player_cruiser,
+              # :player_submarine,
               :computer_cruiser,
               :computer_submarine
 
   def initialize
-    @player_board = Board.new
-    @computer_board = Board.new
+    # @player_board = Board.new
+    # @computer_board = Board.new
     @player_cruiser = Ship.new("Cruiser", 3)
     @player_submarine = Ship.new("Submarine", 2)
-    @computer_cruiser = Ship.new("Cruiser", 3)
-    @computer_submarine = Ship.new("Submarine", 2)
+    # @computer_cruiser = Ship.new("Cruiser", 3)
+    # @computer_submarine = Ship.new("Submarine", 2)
   end
 
-  def setup
+  def game_menu
+    puts "Welcome to BATTLESHIP!"
+    puts "Enter 'p' to play. Enter 'q' to quit."
+    print "> "
+
+    player_input = gets.chomp.downcase
+
+    while player_input != "p" || "q"
```

Comment:
Everything looks great! Can you go a little more in depth on what this while is doing?

Reply:
We had another gets method when we asked to start the game with 'p' so it's just another name for a loop. So if the player input is `'p'` it starts the game, if the input is `'q'` it ends the program (thinking of it now we should add a 'good bye' message or something).. But~ `while` the input isn't `'p' or 'q'` it continues down into the else and just gets another response. It can be done with an until loop as well, something like
```
until player_input == 'p' || 'q'
   if player_input == 'p'```